vignettes/project2_chunks/project2_tim.Rmd

```diff
@@ -65,6 +56,512 @@ if (!require("tidySingleCellExperiment")) {
+
+# plot estimates: 
+library(ggplot2)
+ggplot(tibble(estimate=estimates), aes(estimate)) + 
+  geom_histogram() + 
+  geom_vline(xintercept=sparsity, color="red", lwd=3) +
+  theme_minimal() + 
+  ggtitle("Sparsity of UMI matrix (true value in red)")
+
+# the Central Limit Theorem lives to fight another day,
+# and we have a decent idea of how to navigate our data.
+```
+
+Often, when someone cracks open a single cell dataset, the first thing they do
+is to cluster the cells (based on whatever approach is fashionable). We don't 
+need to do that here. In fact, if you do plot the usual UMAP'ed clusters, 
+you'll mostly just see a batch effect. This takes a while to compute, so I 
+recommend you not bother with the next chunk. I'll drop in a plotly screengrab.
+
+<details>
+  <summary>Only peek if you want to see why UMAP can mislead you</summary>
+```{r, UMAP, eval=FALSE}
+
+library(scater)
+# it is standard to log-normalize counts
+# (although it's not actually a great idea)
+tidybarnyard <- logNormCounts(tidybarnyard)
+
+# compute UMAP embedding on the most variable genes
+tidybarnyard %>% runUMAP(ncomponents=3) -> tidyUMAP
+
+# plot using plotly, color by method
+tidyUMAP %>%
+  plot_ly(
+    x =~ `UMAP1`,
+    y =~ `UMAP2`,
+    z =~ `UMAP3`,
+    color =~ method
+  )
+
+```
+![UMAP plotly](figure/UMAP.png)
+
+</details> 
+
+For this project, we know that the reads from each library (cell) were 
+competitively aligned against the mouse and human genomes. So all we really
+need to do is to decide what we'll call a "mouse cell", what we'll call a 
+"human cell", and which we aren't confident calling either (unclassifiable).
+
+Originally, I wrote a little function to calculate what fraction of transcripts
+from a genome were present in a given cell, and then made a matrix out of the 
+results. But since we already have a place to store the per-cell results,
+why not just use that instead? 
+
+
+```{r, bygenome}
+
+# identify the mouse genes:
+as_tibble(rowData(tidybarnyard)) %>%  # make a tibble from the rowData
+  filter(genome == "mm10") %>%        # select just the mouse mm10 genes 
+  pull("name") -> mouseGenes          # assign their name column to mouseGenes
+
+# for technical reasons, it's faster to tally expressed genes this way:
+tidybarnyard$fracmouse <- 
+  (colSums(counts(tidybarnyard)[mouseGenes, ] > 0)) / length(mouseGenes)
+
+# do the same thing but with human genes:
+as_tibble(rowData(tidybarnyard)) %>%  # make a tibble from the rowData
+  filter(genome == "hg19") %>%        # select just the human hg19 genes 
+  pull("name") -> humanGenes          # assign their name column to humanGenes
+
+# same remarks as previously
+tidybarnyard$frachuman <- 
+  (colSums(counts(tidybarnyard)[humanGenes, ] > 0)) / length(humanGenes)
+
+# why normalize by gene count?
+as_tibble(rowData(tidybarnyard)) %>% select("genome") %>% table
+
+
+# let's use ggplot to make sense of the results:
+barnyardtibble <- as_tibble(tidybarnyard)
```

Comment:
At this spot, I run into this error: Error in as.vector(x) : no method for coercing this S4 class to a vector